docs/ingressroute.md

```diff
@@ -0,0 +1,613 @@
+- Ring hash: The ring/modulo hash load balancer implements consistent hashing to upstream Endpoints
+- Maglev: The Maglev strategy implements consistent hashing to upstream Endpoints
+- Random: The random strategy selects a random healthy Endpoints
+
+More information on the load balancing strategy can be found in [Envoy's documentation](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/load_balancing.html).
+
+The following example IngressRoute defines the strategy for Service `s2-strategy` as `WeightedLeastRequest`.  Service `s1-strategy` does not have an explicit strategy defined so it will use the strategy of `RoundRobin`.
+
+```yaml
+# lb-strategy.ingressroute.yaml
+apiVersion: contour.heptio.com/v1beta1
+kind: IngressRoute
+metadata: 
+  name: lb-strategy
+  namespace: default
+spec: 
+  virtualhost:
+    fqdn: strategy.bar.com
+  routes: 
+    - match: /
+      services: 
+        - name: s1-strategy
+          port: 80
+        - name: s2-strategy
+          port: 80
+          strategy: WeightedLeastRequest
+```
+
+#### IngressRoute Default Load Balancing Strategy (Not supported in beta.1)
+
+In order to reduce the amount of duplicated configuration, the IngressRoute specification supports a default Strategy that will be applied to all Services.  You may still override this default on a per-Service basis.
+
+In this example, Services `s1-def-strategy` and `s2-def-strategy` will both have requests distributed across their Endpoints using the WeightedLeastRequest strategy.  Service `s3-def-strategy` will have requests distributed randomly.
+
+```yaml
+# default-lb-strategy.ingressroute.yaml
+apiVersion: contour.heptio.com/v1beta1
+kind: IngressRoute
+metadata: 
+  name: default-lb-strategy
+  namespace: default
+spec: 
+  virtualhost:
+    fqdn: default-strategy.bar.com
+  strategy: WeightedLeastRequest # Default LB algorithm to be applied to services
+  routes: 
+    - match: /
+      services: 
+        - name: s1-def-strategy
+          port: 80
+        - name: s2-def-strategy
+          port: 80
+        - name: s3-def-strategy
+          port: 80
+          strategy: Random # Overrides default LB algorithm for only this Service
+```
+
+#### Active Health Checking
+
+While Kubernetes' service discovery system provides active health checking through [Pod Readiness Probes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/), there are scenarios where additional active health checking is desired.  For example, when the Kubernetes API server is down (either for regular maintenance or an outage), Pods that are failing Readiness Checks will not be removed from the corresponding Endpoint objects.  Envoy will continue to try to route traffic to these failing pods which results in user-facing HTTP errors or timeouts.
```

Comment:
Should this section describe Liveliness Probes instead of Readiness?

Reply:
I believe I'm correct in referencing Readiness Probes. 

> The kubelet uses liveness probes to know when to restart a Container. 
> The kubelet uses readiness probes to know when a Container is ready to start accepting traffic.

Reply:
Readiness only happens at the start of the container, once it passes it's never called again. Liveliness is called continually as the container runs to make sure it's still healthy.

Reply:
Steve's CKA and CKAD have been revoked :)

https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#define-readiness-probes